### PR TITLE
Show/Hide bug fix

### PR DESCRIPTION
**Pull request by onursumer**
# What? Why?
Fixes and improves the show/hide feature behavior.

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

**Comment by onursumer** [comment]
[(3) fix show/hide](https://trello.com/c/6TQmq9rC/86-3-fix-show-hide)

**Comment by onursumer** [comment]
@alisman @adamabeshouse The fix is actually one line to make the show/hide work again: https://github.com/cBioPortal/cbioportal-frontend/pull/144/files#diff-dd74d85f67f0228723795bf4b6f21e1dR497

But I also made some improvements. Basically, now the `generateRows` method is called only when the column definitions are updated (since we are passing additional data via `columnProps` within the column definition). Change in `columnVisibility` now only affects `generateHeaders`, not `generateRows` anymore. Updating the `Thead` is sufficient to show/hide columns, since reactable is smart enough to ignore `Td` components with an unknown `column` value.

File: src/shared/components/enhancedReactTable/EnhancedReactTable.tsx [hunk 1]
```diff
@@ -106,33 +93,42 @@ export default class EnhancedReactTable<T> extends React.Component<IEnhancedReac
...
 
     constructor(props:IEnhancedReactTableProps<T>)
     {
         super(props);
 
+        this.colNameToId = this.mapColNameToId(props.columns);
+
         this.state = {
             columnVisibility: EnhancedReactTable.resolveVisibility(props.columns, props.rawData),
+            columnSort: this.resolveSortable(this.props.columns || {}),
```
Comment (adamabeshouse) [inline src/shared/components/enhancedReactTable/EnhancedReactTable.tsx]: this name doesn't make sense to me/is sort of vague. It's not quite a verb, not quite a noun. I think it should be sortableColumns or something. same with columnFilter

Comment (onursumer) [inline src/shared/components/enhancedReactTable/EnhancedReactTable.tsx]: okay, let's change to `sortableColumns` and `filterableColumns`.

**Pull request merged**